kolibri/plugins/coach/assets/src/views/plan/LessonSummaryPage/index.vue

```diff
@@ -207,6 +232,61 @@
           }
         }
       },
+      showResourcesRemovedNotification() {
+        const undo = () => {
+          this.save(this.workingResourcesBackup);
+          this.clearSnackbar();
+        };
+        const hide = () => {
+          if (this.workingResourcesBackup) {
+            // snackbar might carryover to another page (like select)
+            this.workingResourcesBackup = [...this.workingResources];
+          }
+        };
+        this.showSnackbarNotification(
+          'resourcesRemovedWithCount',
+          { count: this.numberOfRemovals },
+          {
+            autoDismiss: true,
+            duration: REMOVAL_SNACKBAR_TIME,
+            actionText: this.$tr('undoActionPrompt'),
+            actionCallback: undo,
+            hideCallback: hide,
+          },
+        );
+      },
+      async handleResourcesChange({ newArray }) {
```

Comment:
All of the work here to handle the resource management within the table, while certainly implied by/part of the spec, is tough, and you did a great job with it. It's a lot more complicated than it would seem just by looking at the UI. And I really like the addition of the snackbar here and the `undo` action. It adds complexity but a really thoughtful implementation for the user perspective.